pkg/visor/peering_daemon.go

```diff
@@ -0,0 +1,109 @@
+package visor
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net"
+	"net/rpc"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/SkycoinProject/dmsg/cipher"
+	skycoin_cipher "github.com/SkycoinProject/skycoin/src/cipher"
+	"github.com/SkycoinProject/skycoin/src/util/logging"
+	spd "github.com/SkycoinProject/skywire-peering-daemon/pkg/daemon"
+	"github.com/rjeczalik/notify"
+
+	"github.com/SkycoinProject/skywire-mainnet/pkg/snet"
+)
+
+var (
+	rpcAddr         = "localhost:3435"
+	logger          = logging.MustGetLogger("SPD")
+	rpcDialTimeout  = time.Duration(5 * time.Second)
+	rpcConnDuration = time.Duration(60 * time.Second)
+	spdMu           sync.Mutex
+)
+
+func execute(cmd *exec.Cmd) error {
+	cmd.Stdout = os.Stdout
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func client() (RPCClient, error) {
+	conn, err := net.DialTimeout("tcp", rpcAddr, rpcDialTimeout)
+	if err != nil {
+		return nil, fmt.Errorf("RPC connection failed: %s", err)
+	}
+	if err := conn.SetDeadline(time.Now().Add(rpcConnDuration)); err != nil {
+		return nil, fmt.Errorf("RPC connection failed: %s", err)
+	}
+	return NewRPCClient(rpc.NewClient(conn), RPCPrefix), nil
+}
+
+// transport establshes an stcp transport to a remote visor
+func createTransport(networkType string, packet spd.Packet) (*TransportSummary, error) {
```

Comment:
Why is the second input of type `spd.Packet` instead of `cipher.PubKey`?

Reply:
Passing any of `spd.Packet` or `cipher.PubKey` would work, and I would only need to change a line in the function body to suit the passed parameter. But looking at it again I think passing `cipher.Pubkey` is simpler.

Reply:
@Kifen Also in the comment it states `stcp transport`, but the first input is `networkType`.

Reply:
True. The `networkType` parameter is redundant.